text/0031-corejam.md

```diff
@@ -0,0 +1,644 @@
+2. Anyone wanting to create decentralised/unstoppable/resilient applications.
+3. Teams already building on Polkadot.
+
+## Explanation
+
+**CoreJam is a general model for utilization of Polkadot Cores. It is a mechanism by which Work Packages are communicated, authorized, computed and verified, and their results gathered, combined and accumulated into particular parts of the Relay-chain's state.**
+
+### From Old to New
+
+The current specification of the Polkadot protocol, and with in the Relay-chain operation, is designed in line with the overall requirements of and terminology in the Polkadot (1.0) whitepaper. It incorporates first-class concepts including *Proof-of-Validity* and *Parachain Validation Function*. This will no longer be considered canonical for the Polkadot protocol. To avoid confusion, this design will be known as the *Fixed-function Parachains Legacy Model*, or the *Legacy Model* for short.
+
+Existing functionality relied upon by parachains will continue to be provided as a special case under a more general and permissionless model which is detailed presently and known as *CoreJam*. Transition of Polkadot to be in line with the present proposal will be necessarily imply some minor alterations of formats utilized by Cumulus, Smoldot and other light-client APIs (see the section on Compatibility). However, much of the underlying logic (in particular, consensus, disputes and availability) is retained, though its application is generalised. This proposal will only make note of the expectations regarding the changes, and presumes continuation of all other logic.
+
+As part of this model, we introduce a number of new and interrelated concepts: *Work Package*, *Work Class*, *Work Item*, *Work Output*, *Work Result*, *Work Report* and *Work Package Attestation*, *Work Class Trie*.
+
+Focussing on continuity and reuse of existing logic, it is unsurprising that many of these concepts already analogues in the Parachains model, albeit ones with a less general definition. While this mapping can be helpful to quickly create an approximate understanding of the new concepts for those already familiar with Polkadot, care must be taken not to inadvertantly make incorrect presumptions over exact details of their relationships, constraints, timing, provisions and APIs. Nonetheless, they are provided here for whatever help they may be.
+
+| CoreJam model                | Legacy model      | Context |
+| --- | --- | --- |
+| *Work Package*               | Proof-of-Validity | Untrusted data provided to RcVG |
+| *Work Item*                  | Proof-of-Validity | State-transition inputs and witness |
+| *Work Output*                | Candidate         | State-transition consequence |
+| *Work Report*                | Candidate         | Target of Attestation |
+| *(Work Package) Attestation* | Attestation       | Output signed by members of RcVG |
+| *Reporting*                  | Attestation       | Placement of Attestation on-chain |
```

Comment:
I forget how attestation gets used now, but I vaguely recall it being the on-chain part.  Anyways..

We'd should preserve the word "backing" for the off-chain part, because this helps explain security.

We do however have an ugly mess in how backing appears in the on-chain part.   I'd suggest: 
- "Backing attestation" is what the backing checker does.
- "Backing report" is when it appears on-chain.

Backing attestations only make someone responsible, but backing and availability provide no security themselves, only approval votes provide security.

Reply:
The ins and outs of the underlying security-game-theory are not of primary relevant within this RFC since it doesn't touch any of that apparatus.